seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py

```diff
@@ -0,0 +1,180 @@
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ASRSmaldusc(datasets.GeneratorBasedBuilder):
+    """ASR-Smaldusc consists transcribed Malay scripted speech focusing on daily use sentences."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "sptext"
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "channel": datasets.Value("string"),
+                    "uttrans_id": datasets.Value("string"),
+                    "speaker_id": datasets.Value("string"),
+                    "transcription": datasets.Value("string"),
+                    "path": datasets.Value("string"),
+                    "audio": datasets.Audio(sampling_rate=16_000),
+                    "speaker_gender": datasets.Value("string"),
+                    "speaker_age": datasets.Value("int64"),
+                    "speaker_region": datasets.Value("string"),
+                    "speaker_device": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.speech_text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        data_paths = {
+            _DATASETNAME: Path(dl_manager.download_and_extract(_URLS[_DATASETNAME])),
+        }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_paths[_DATASETNAME],
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        # read UTTRANSINFO file
+        # columns: channel, uttrans_id, speaker_id, prompt (empty field), transcription
```

Comment:
Any reason why do you omit the `prompt` field for the `source` schema? It's not empty and can be accessed based on your code using `uttransinfo_data[i][3]`.

Reply:
I meant the prompts are empty strings

Reply:
It's not? This is what I got running your code.

```python
>>> import os
>>> from pathlib import Path
>>> from datasets import DownloadManager
>>> URL = "https://magichub.com/df/df.php?file_name=Malay_Scripted_Speech_Corpus_Daily_Use_Sentence.zip"
>>> filepath = Path(DownloadManager().download_and_extract(URL))
>>> uttransinfo_filepath = os.path.join(filepath, "UTTRANSINFO.txt")
>>> with open(uttransinfo_filepath, "r", encoding="utf-8") as uttransinfo_file:
...     uttransinfo_data = uttransinfo_file.readlines()
... 
>>> uttransinfo_data = uttransinfo_data[1:]  # remove header
>>> uttransinfo_data = [s.strip("\n").split("\t") for s in uttransinfo_data]
>>> uttransinfo_data[:5]
[['C1', 'G5001_1_S0002.wav', 'G5001', 'Dia tidak tahu apa-apa .', 'dia tidak tahu apa-apa'], ['C1', 'G5001_1_S0003.wav', 'G5001', 'Dia segera beredar dari situ .', 'dia segera beredar dari situ'], ['C1', 'G5001_1_S0005.wav', 'G5001', 'Dia menoleh ke arah gadis itu .', 'dia menoleh ke arah gadis itu'], ['C1', 'G5001_1_S0006.wav', 'G5001', 'Lelaki itu memandangnya dari atas ke bawah .', 'lelaki itu memandang dari atas ke bawah'], ['C1', 'G5001_1_S0007.wav', 'G5001', 'Dia menarik nafas dalam-dalam dan menghembuskannya dengan perlahan .', 'dia menarik nafas dalam-dalam dan menghembuskannya dengan perlahan']]
```

Reply:
I checked again and yes you are right, it does have the prompt. I was looking at another similar dataset. I've added the prompt field and pushed the change. Thanks for catching.